kernel/include/ksched.h

```diff
@@ -383,16 +383,34 @@ int z_sched_wait(struct k_spinlock *lock, k_spinlock_key_t key,
  */
 void z_sched_usage_stop(void);
 
-void z_sched_usage_start(struct k_thread *thread);
+/**
+ * @brief Start thread cycle usage accounting
+ *
+ * This routine is called to mark the beginning of thread cycle usage
+ * and accounting.
+ *
+ * @param thread Identifies thread for which to track cycles
+ * @param new_window False unless a new thread is being switched in
+ */
+void z_sched_usage_start(struct k_thread *thread, bool new_window);
```

Comment:
I'm looking, and I don't see anywhere this gets called without a static "1" for the second argument?  Did something get forgotten?  I'm guessing maybe it's a parallel mixup to mine earlier, confusing the start of a thread scheduling quantum with the start of a "cpu usage window"?  As it stands, it looks like indeed every interrupt is going to split the windows, no?

Reply:
Great catch!  It was only supposed to get called with "1" (or "true" in one place).

Must definitely fix.

Reply:
I went back through my notes, and this was actually a leftover artefact from a previous idea.

A new window occurs whenever a new thread is scheduled in. Unless I am missing something, this is not something that should be occurring every interrupt (unless a new thread is getting scheduled to run after every interrupt).